apiserver/pkg/modelservice/modelservice.go

```diff
@@ -0,0 +1,325 @@
+		Namespace:         obj.GetNamespace(),
+		CreationTimestamp: &creationTimestamp,
+		Labels:            graphqlutils.MapStr2Any(obj.GetLabels()),
+		Annotations:       graphqlutils.MapStr2Any(obj.GetAnnotations()),
+		DisplayName:       &embedder.Spec.DisplayName,
+		Description:       &embedder.Spec.Description,
+		Type:              &serviceType,
+		Provider:          &provider,
+		BaseURL:           baseURL,
+		Models:            embedder.GetModelList(),
+		Status:            &status,
+		Message:           &message,
+		UpdateTimestamp:   &updateTime,
+	}
+	return &md
+}
+
+func obj2LLM(ctx context.Context, c dynamic.Interface, obj *unstructured.Unstructured) *generated.Llm {
+	llm := &v1alpha1.LLM{}
+	if err := utils.UnstructuredToStructured(obj, llm); err != nil {
+		return &generated.Llm{}
+	}
+
+	id := string(llm.GetUID())
+	creationTimestamp := llm.GetCreationTimestamp().Time
+
+	// conditioned status
+	condition := llm.Status.GetCondition(v1alpha1.TypeReady)
+	updateTime := condition.LastTransitionTime.Time
+	status := string(condition.Status)
+	message := string(condition.Message)
+
+	llmType := string(llm.Spec.Type)
+	provider := string(llm.Spec.Provider.GetType())
+
+	// get llm's api url
+	var baseURL string
+	switch llm.Spec.Provider.GetType() {
+	case v1alpha1.ProviderTypeWorker:
+		baseURL, _ = common.GetAPIServer(ctx, c, true)
+	case v1alpha1.ProviderType3rdParty:
+		baseURL = llm.Spec.Enpoint.URL
+	}
+
+	md := generated.Llm{
+		ID:                &id,
+		Name:              obj.GetName(),
+		Namespace:         obj.GetNamespace(),
+		CreationTimestamp: &creationTimestamp,
+		Labels:            graphqlutils.MapStr2Any(obj.GetLabels()),
+		Annotations:       graphqlutils.MapStr2Any(obj.GetAnnotations()),
+		DisplayName:       &llm.Spec.DisplayName,
+		Description:       &llm.Spec.Description,
+		Type:              &llmType,
+		Status:            &status,
+		Message:           &message,
+		Provider:          &provider,
+		BaseURL:           baseURL,
+		Models:            llm.GetModelList(),
+		UpdateTimestamp:   &updateTime,
+	}
+	return &md
+}
+
+func CreateModelService(ctx context.Context, c dynamic.Interface, input generated.CreateModelServiceInput) (*generated.ModelService, error) {
+	// Overview:
+	// - Get general info from input: displayName, description, types, name & namespace, etc.
+	// - Create *generated.LLM, *generated.embedder accordingly
+	// - Wrap all elements into *generated.ModelService
+	displayName, description, serviceType, APIType := "", "", "", ""
+	var genLLM *generated.Llm
+	var genEmbed *generated.Embedder
+
+	if input.DisplayName != nil {
+		displayName = *input.DisplayName
+	}
+	if input.Description != nil {
+		description = *input.Description
+	}
+	if input.Types != nil {
+		serviceType = *input.Types
+	}
+	if input.APIType != nil {
+		APIType = *input.APIType
+	}
+
+	// FIXME: Codes duplicated with llm & embedder creation.
+	if strings.Contains(serviceType, "llm") {
+		llm := v1alpha1.LLM{
```

Comment:
Please move this LLM creation code to https://github.com/kubeagi/arcadia/blob/main/apiserver/pkg/llm/llm.go just like 
https://github.com/kubeagi/arcadia/blob/main/apiserver/pkg/embedder/embedder.go#L87